trinity/plugins/builtin/tx_pool/plugin.py

```diff
@@ -0,0 +1,87 @@
+from argparse import (
+    ArgumentParser
+)
+import asyncio
+
+from evm.chains.base import (
+    BaseChain
+)
+from evm.chains.mainnet import (
+    BYZANTIUM_MAINNET_BLOCK,
+    MainnetChain,
+)
+from evm.chains.ropsten import (
+    BYZANTIUM_ROPSTEN_BLOCK,
+    RopstenChain,
+)
+
+from p2p.cancel_token import (
+    CancelToken
+)
+from p2p.peer import (
+    PeerPool
+)
+
+from trinity.extensibility import (
+    BaseEvent,
+    BasePlugin,
+    PluginContext,
+)
+from trinity.extensibility.events import (
+    ResourceAvailableEvent,
+    TrinityStartupEvent,
+)
+from trinity.plugins.builtin.tx_pool.pool import (
+    TxPool,
+)
+from trinity.plugins.builtin.tx_pool.validators import (
+    DefaultTransactionValidator
+)
+
+
+class TxPlugin(BasePlugin):
+
+    def __init__(self) -> None:
+        self.peer_pool: PeerPool = None
+        self.cancel_token: CancelToken = None
+        self.chain: BaseChain = None
+        self.is_enabled: bool = None
+
+    @property
+    def name(self) -> str:
+        return "TxPlugin"
+
+    def configure_parser(self, arg_parser: ArgumentParser) -> None:
+        arg_parser.add_argument(
+            "--disable-tx-pool",
+            action="store_true",
+            help="Disable the Transaction Pool",
+        )
+
+    def handle_event(self, activation_event: BaseEvent) -> None:
+        if isinstance(activation_event, TrinityStartupEvent):
+            self.is_enabled = not activation_event.args.disable_tx_pool
+        if isinstance(activation_event, ResourceAvailableEvent):
+            if activation_event.resource_type is PeerPool:
+                self.peer_pool, self.cancel_token = activation_event.resource
+            elif activation_event.resource_type is BaseChain:
+                self.chain = activation_event.resource
+
+    def should_start(self) -> bool:
+        return all(
+            val is not None and val is not False
+            for val in [self.peer_pool, self.chain, self.is_enabled]
```

Comment:
I think this is cleaner as:

```python
return all((self.peer_pool, self.chain, self.is_enabled))

# or
if not self.peer_pool:
    return False
elif not self.chain:
    return False
elif not self.is_enabled:
    return False
else:
    return True
```

Either would be 👍 from me.

Reply:
The first one looks good. I just learned that Python has the concept of Truthy/Falsey as well. Thought that's a JavaScript thing only :+1:

Reply:
Oh turns out, `all(...)` does expect a proper `boolean` so that doesn't work. However, it can be shortened to this:

```python
        return all((self.peer_pool != None, self.chain != None, self.is_enabled))
```

Reply:
> all(...) does expect a proper boolean

Truth-y things work in `all()`. See:

```py
assert all([object(), 'tru', 1, True])
```